.azure-pipelines/impacted_area_testing/get-impacted-area.yml

```diff
@@ -0,0 +1,76 @@
+steps:
+- script: |
+    set -x
+
+    DIFF_FOLDERS=$(git diff HEAD^ HEAD --name-only | xargs -n1 dirname | sort -u | tr '\n' ' ')
+
+    if [[ $? -ne 0 ]]; then
+      echo "##vso[task.complete result=Failed;]Get diff folders fails."
+      exit 1
+    else
+      echo -n "##vso[task.setvariable variable=DIFF_FOLDERS]$DIFF_FOLDERS"
+    fi
+  continueOnError: false
+  displayName: "Get diff folders"
+
+- script: |
+    set -x
+
+    pip install PyYAML
+    pip install natsort
+
+    sudo apt-get install -y jq
+
+    FINAL_FEATURES=""
+    IFS=' ' read -ra FEATURES_LIST <<< "$(DIFF_FOLDERS)"
+    for FEATURE in "${FEATURES_LIST[@]}"
+    do
+      # If changes contains the common part in tests folder,the scope of PR testing is all test scripts.
+      if [[ "$FEATURE" == *tests/common* ]]; then
+        FINAL_FEATURES=""
+        break
+
+      # If changes only limited to specific feature, the scope of PR testing is impacted area.
+      elif [[ "$FEATURE" =~ tests\/* ]]; then
+        # Cut the feature path
+        if [[ $FEATURE == */*/* ]]; then
+            FEATURE=$(echo "$FEATURE" | cut -d'/' -f1-2)
+        fi
+
+        if [[ -z "$FINAL_FEATURES" ]]; then
+          FINAL_FEATURES="${FEATURE#tests/}"
+        else
+          FINAL_FEATURES="$FINAL_FEATURES,${FEATURE#tests/}"
+        fi
+
+      # If changes related to other folders excpet tests, we also consider them as common part.
+      # The scope of PR testing is all test scripts.
+      else
+        FINAL_FEATURES=""
+        break
+      fi
+    done
+
+    if [[ -z "$FINAL_FEATURES" ]]; then
+      TEST_SCRIPTS=$(python ./.azure-pipelines/impacted_area_testing/get_test_scripts.py --features "" --location tests)
+    else
+      TEST_SCRIPTS=$(python ./.azure-pipelines/impacted_area_testing/get_test_scripts.py --features ${FINAL_FEATURES} --location tests)
+    fi
```

Comment:
Here do not need if-else

Reply:
done

Reply:
`if [[ -z "$FINAL_FEATURES" ]];` doesn't mean $FINAL_FEATURES is `""`. If we remove the if-else here, and change the code to 
```TEST_SCRIPTS=$(python ./.azure-pipelines/impacted_area_testing/get_test_scripts.py --features ${FINAL_FEATURES} --location tests)
```
We will get below error:
```
python ./.azure-pipelines/impacted_area_testing/get_test_scripts.py --features --location tests
usage: get_test_scripts.py [-h] [--features FEATURES] [--location LOCATION]
get_test_scripts.py: error: argument --features: expected one argument
```

Reply:
yeah, you can add python parse params to support no value or None value passing.
```
nargs="?", 
const="", 
```
Such as test_plan.py: 
![image](https://github.com/user-attachments/assets/53a45ea6-c249-409e-9e8b-c21332a98f45)
https://github.com/sonic-net/sonic-mgmt/blob/ca01ce6eb2cb97a2bddd3493c457c6b64af18a22/.azure-pipelines/test_plan.py#L765

Reply:
I think, this change can be applied in our second stage of rolling out.
In the first stage, we only focus on the changes which only modify the feature folders.

Reply:
And I will modify the code for the first stage of rolling out.